helsingborg-stad/aterbruket-webapp
# Wip/filter option

### **Explain the changes you've made**

- fixed the filtering function for the category (filtered through graphQl) and condition (filtered based on the fetched items(already filtered with category).

- changed "ny" to "nyskick".  

- put "condition" in the card instead of "status" (on the view still in english, will change later to swedish)

- added  no item fits the filter criteria message ! (not implemented design yet)

- fixed the bug that occurred when only filter in condition several times

### **Explain why these changes are made**

Because the filtering function was not filtering both category and condition.

### **Explain your solution**

In order to achieve multiple filtering in graphql the query structure needs to be altered but it is not feasible at the moment. So In order to make the project go further, now I fixed it by combing filtering once in graphql and filtering again in front end.
This filtering function will be reviewed again when backender joins. 

### **How to test the changes?**

go to home page, click on the filter area to see the filter result, and test on:   

1. when items fits the filter criteria in both category and condition
2. only filter with category
3. only filter with condition
4. filter for "none items fits the criteria"
